clang/lib/Basic/Targets/Xtensa.cpp

```diff
@@ -0,0 +1,62 @@
+//===--- Xtensa.cpp - Implement Xtensa target feature support -------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements Xtensa TargetInfo objects.
+//
+//===----------------------------------------------------------------------===//
+
+#include "Xtensa.h"
+#include "clang/Basic/Builtins.h"
+#include "clang/Basic/MacroBuilder.h"
+#include "clang/Basic/TargetBuiltins.h"
+
+using namespace clang;
+using namespace clang::targets;
+
+void XtensaTargetInfo::getTargetDefines(const LangOptions &Opts,
+                                        MacroBuilder &Builder) const {
+  Builder.defineMacro("__xtensa__");
+  Builder.defineMacro("__XTENSA__");
+  if (BigEndian)
```

Comment:
On targets which have two endian variants; the endianness is usually part of the triple, because there isn't any ABI-compatible subset.

I won't insist on it for an embedded target like this, but you might want to consider it.

Reply:
Sorry. Do you mean to support for `-triple=xtensale` and `-triple=xtensaeb` and have corresponding checks in `clang/test/Preprocessor/init.c`?

Reply:
I can do like it is done in ARM: `xtensa` - little-endian and `xtensaeb` - big-endian. But for big endian support I will have to update LLVM Triple code also. If that will be acceptable to be a part of this Clang related patch I will do it.

Another option is to support `xtensa` only and hard-code `BigEndian = false` until the patch with big endianess support if any.

Reply:
I think it makes sense to just hardcode `BigEndian = false` for now, and revisit in a followup patch.

Something like xtensaeb is what I was thinking, yes.

Reply:
fixed